src/VisualStudio/IntegrationTest/TestUtilities/InProcess/Editor_InProc.cs

```diff
@@ -501,5 +502,43 @@ protected override ITextBuffer GetBufferContainingCaret(IWpfTextView view)
         {
             return view.GetBufferContainingCaret();
         }
+
+        public List<string> GetF1Keyword()
+        {
+            return InvokeOnUIThread(() =>
+            {
+                List<string> results = new List<string>();
+                IVsTextLines textLines;
+                GetActiveVsTextView().GetBuffer(out textLines);
+                Guid languageServiceGuid;
+                textLines.GetLanguageServiceID(out languageServiceGuid);
+                object languageService;
+                Microsoft.VisualStudio.Shell.ServiceProvider.GlobalProvider.QueryService(languageServiceGuid, out languageService);
+                var languageContextProvider = languageService as IVsLanguageContextProvider;
+
+                IVsMonitorUserContext monitorUserContext = GetGlobalService<SVsMonitorUserContext, IVsMonitorUserContext>();
+                IVsUserContext emptyUserContext;
+                monitorUserContext.CreateEmptyContext(out emptyUserContext);
```

Comment:
Can you observe the results of these calls and give nice failure exceptions if things aren't `S_OK`?